File: Valghalla.Internal.Web/src/features/administration/task-type/models/create-task-type-request.ts
export interface CreateTaskTypeRequest {
  title: string;
  shortName: string;
  description: string;
  startTime: string;
  endTime: string;
  payment?: number;
  validationNotRequired: boolean;
  trusted: boolean;
  sendingReminderEnabled: boolean;
  fileReferenceIds: string[];
}

File: Valghalla.Internal.API/Requests/Administration/TaskType/CreateTaskTypeRequest.cs
namespace Valghalla.Internal.API.Requests.Administration.TaskType
{
    public sealed record CreateTaskTypeRequest
    {
        public string Title { get; init; } = null!;
        public string ShortName { get; init; } = null!;
        public string Description { get; init; } = null!;
        public TimeSpan StartTime { get; init; }
        public TimeSpan EndTime { get; init; }
        public int? Payment { get; init; }
        public bool ValidationNotRequired { get; init; }
        public bool Trusted { get; init; }
        public bool SendingReminderEnabled { get; init; } = true;
        public IEnumerable<Guid> FileReferenceIds { get; init; } = Enumerable.Empty<Guid>();
    }
}
